src/cluster.bootstrap/Akka.Management.Cluster.Bootstrap/README.md

```diff
@@ -0,0 +1,280 @@
+# Akka.Management.Cluster.Bootstrap
+Akka Cluster Bootstrap helps forming (or joining to) a cluster by using Akka Discovery to discover peer nodes. 
+It is an alternative to configuring static seed-nodes in dynamic deployment environments such as on Kubernetes or AWS.
+
+It builds on the flexibility of Akka Discovery, leveraging a range of discovery mechanisms depending on the environment you 
+want to run your cluster in.
+
+Cluster bootstrap depends on:
+- Akka.Discovery to discover other members of the cluster
+- Akka.Management to host HTTP endpoints used during the bootstrap process
+
+## Usage
+Akka management must be started as well as the bootstrap process, this can either be done through config or programmatically.
+
+### Setting Up Cluster.Bootstrap from HOCON Configuration
+
+Listing the ClusterBootstrap extension among the autoloaded akka.extensions in your configuration will cause it to autostart:
+
+```
+# trigger autostart by loading the extension through config
+akka.extensions = ["Akka.Management.Cluster.Bootstrap.ClusterBootstrapProvider, Akka.Management.Cluster.Bootstrap"]
+```
+
+If management or bootstrap configuration is incorrect, the autostart will log an error and terminate the actor system.
+
+### Setting Up Cluster.Bootstrap Programmatically
+```
+// Akka Management hosts the HTTP routes used by bootstrap
+await AkkaManagement.Get(system).Start()
+
+// Starting the bootstrap process needs to be done explicitly
+await ClusterBootstrap.Get(system).start()
+```
+
+Ensure that `seed-nodes` is not present in configuration and that either autoloading through config or start() is called on every node.
+
+The following configuration is required, more details for each and additional configuration can be found in the reference configuration:
+
+- `akka.management.cluster.bootstrap.contact-point-discovery.service-name`: a unique name in the deployment environment for this 
+  cluster instance which is used to lookup peers in service discovery. If unset, it will be derived from the ActorSystem name.
+- `akka.management.cluster.bootstrap.contact-point-discovery.discovery-method`: the intended service discovery mechanism 
+  (from what choices Akka Discovery provides). If unset, falls back to the system-wide default from akka.discovery.method.
```

Comment:
Should link to relevant Akka.Discovery documentation here: https://getakka.net/articles/discovery/index.html - although that documentation is likely out of date and incomplete as well.